### PR TITLE
source:xlsx added the xlsx source

### PR DESCRIPTION
**Pull request by Pratikrocks**
Signed-off-by: Pratik Dey <pratikrocks.dey11@gmail.com>

### CODE REVIEW

File: source/xlsx/dffml_source_xlsx/xlsx.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+@entrypoint("xlsx")
+class XLSXSource(FileSource, MemorySource):
+    """
+    Source to read files in .xlsx format.
+    """
+
+    CONFIG = XLSXSourceConfig
+
+    async def load_fd(self, fd):
+        excel_file = load_workbook(fd.name)  # Loading the Workbook
```
Comment (Pratikrocks) [inline source/xlsx/dffml_source_xlsx/xlsx.py:26]: @pdxjohnny added the functionality for `load_fd`

File: source/xlsx/dffml_source_xlsx/xlsx.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    row=row, column=col
+                )  # Cell content corresponding to the feature data
+                temp_dict[str(feature.value)] = cell_element.value
+            self.mem[str(row - 1)] = Record(
+                str(row - 1), data={"features": temp_dict}
+            )
+        self.logger.debug("%r loaded %d records", self, len(self.mem))
+
+    async def dump_fd(self, fd):
+        excel_file = Workbook()
```
Comment (Pratikrocks) [inline source/xlsx/dffml_source_xlsx/xlsx.py:50]: functionality for `dump_fd`

File: source/xlsx/dffml_source_xlsx/xlsx.py (hunk 1 above)
Comment (Pratikrocks) [reply source/xlsx/dffml_source_xlsx/xlsx.py:26]: Please suggest if some additional modifications are required

File: source/xlsx/xlsx.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+from openpyxl import Workbook, load_workbook
```
Comment (pdxjohnny) [inline source/xlsx/xlsx.py:1]: Let's remove this file

File: source/xlsx/dffml_source_xlsx/xlsx.py (hunk 1 above)
Comment (pdxjohnny) [reply source/xlsx/dffml_source_xlsx/xlsx.py:26]: Let's add a `if not hasattr(fd.name): raise NotImplementedError(...)` in addition to creating the issue and the `# TODO` comment

Comment (Pratikrocks) [reply source/xlsx/dffml_source_xlsx/xlsx.py:26]: Yes we can have it , updating 👍

**Pull request closed without merging**